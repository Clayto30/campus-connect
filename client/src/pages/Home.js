import React from 'react';
import { useAuthState } from '../../utils/auth';
import { Typography, makeStyles } from '@material-ui/core';

const Home = () => {
	const { user } = useAuthState();
	console.log(user);
	return (
		<div>
			{!user ? (
				<>
					<Typography>Welcome to Campus Connect</Typography>
				</>
			) : (
				<>
					<Typography>Welcome {user.data.username}</Typography>
				</>
			)}
		</div>
	);
};

export default Home;
