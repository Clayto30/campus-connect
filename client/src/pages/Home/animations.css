.bounce-in {
	text-align: center;
	font-size: 7rem;
	animation: bounce-in 1s ease-in-out 0s 1 normal;
}

@-webkit-keyframes bounce-in {
	0% {
		opacity: 0;
		transform: scale(0.3);
	}
	50% {
		opacity: 1;
		transform: scale(1);
	}
	70% {
		transform: scale(0.9);
	}
	100% {
		transform: scale(1);
	}
}

@keyframes bounce-in {
	0% {
		opacity: 0;
		transform: scale(0.3);
	}
	50% {
		opacity: 1;
		transform: scale(1);
	}
	70% {
		transform: scale(0.9);
	}
	100% {
		transform: scale(1);
	}
}
