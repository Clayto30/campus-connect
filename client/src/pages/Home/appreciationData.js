const appreciationData = {
    appreciation: [
        {
            to: "Team Campus Connect",
            message: `Many Thanks to all of your hard work and dedication over the last couple of weeks. This being the hardest group project on all of boot camp
            we did well. Lets cheer ourselves and give a pat on your back for job well done`,
            from: "Janarthani",
            imageUrl: "https://acegif.com/wp-content/gif/wowo-applause-10.gif"
        },
        {
            to: "Manny",
            message: "East or West Manny is the best. Appreciate all you help and support throught this course. You have been awesome.",
            from: "Team Campus Connect",
            imageUrl: "https://acegif.com/wp-content/gif/wowo-applause-10.gif"
        },
        {
            to: "Aliff",
            message: "You are our Hero. Thank you for coming to our rescue Aliff. We knew we could count on you",
            from: "Team Campus Connect",
            imageUrl: "https://acegif.com/wp-content/gif/wowo-applause-10.gif"
        },
        {
            to: "Nate",
            message: "Thank you for extending a helping hand whenever we needed help. Without you we wouldnt have had the progress we did",
            from: "Team Campus Connect",
            imageUrl: "https://acegif.com/wp-content/gif/wowo-applause-10.gif"
        },
        {
            to: "Mason",
            message: "Thank you for all you motivation and pep talks Mason. Appreicate your detailed feedback on projects",
            from: "Team Campus Connect",
            imageUrl: "https://acegif.com/wp-content/gif/wowo-applause-10.gif"
        },
        {
            to: "Wesley",
            message: "What would be do without you Wesley. Appreciate you nudding in the right direction whenever we felt lost",
            from: "Team Campus Connect",
            imageUrl: "https://acegif.com/wp-content/gif/wowo-applause-10.gif"
        },
        {
            to: "Amanda",
            message: "You worked your magic and got rid of the bugs everytime. Appreciate all your help Amanda",
            from: "Team Campus Connect",
            imageUrl: "https://acegif.com/wp-content/gif/wowo-applause-10.gif"
        },

    ]

}

export default appreciationData;