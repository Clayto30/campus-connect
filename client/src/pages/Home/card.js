import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles((theme) => ({
    carddiv: {
        maxWidth: 545,
        width: 240,
        margin: 10,
        '@media (min-width:600px)': {
            width: 590,
        },
    },
    para: {
        fontFamily: `roboto`,
        color: '#003262',
        lineHeight: '1.5',
        fontWeight: 300,
        fontSize: '1rem',
        textAlign: 'center',
        margin: theme.spacing(2),
        '@media (min-width:600px)': {
            fontSize: '1.2rem',
        },
        '@media (min-width:1200px)': {
            fontSize: '1.4rem',
        },
    },
    media: {
        height: '350',
        borderRadius: '4%',
    },
    title: {
        fontSize: '1rem',
        textAlign: 'center',
        paddingTop: '2rem',
        color: "#003262",
        '@media (min-width:600px)': {
            fontSize: '1.5rem',
        },
    },
    signature: {
        textAlign: 'right',
        color: '#003262',
        fontFamily: 'cursive',
    },
}));

const Appreciate = ({ to, imageUrl, message, from }) => {
    const classes = useStyles();

    return (
        <Card className={classes.carddiv}>
            <img
                alt="funny gif"
                className={classes.media}
                src={imageUrl}
            />
            <CardContent className={classes.cardbody}>
                <Typography gutterBottom className={classes.title} component="h2">
                    Dear {to}
                </Typography>
                <Typography component="p" className={classes.para} >
                    {message}
                </Typography>
                <Typography component="p" className={classes.signature}>
                    {from}
                </Typography>
            </CardContent>

        </Card>
    )
}

export default Appreciate

