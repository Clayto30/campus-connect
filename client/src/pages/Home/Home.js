import React from 'react';
import './animations.css';
import { useAuthState } from '../../utils/auth';
import { makeStyles } from '@material-ui/core/styles';
import { Typography, Container } from '@material-ui/core';
import appreciationData from './appreciationData';
import Appreciate from './card';

const Home = props => {
	const useStyles = makeStyles((theme) => ({
		celebrate: {
			marginTop: '50%',
		},
		root: {
			flexGrow: 1,
			width: '100%',
			display: 'flex',
			justifyContent: 'center',
			flexWrap: 'wrap',
		},
		slide: {
			display: 'flex',
			justifyContent: 'center',
			alignItems: 'center',
			width: '40%',
		}
	}));

	const classes = useStyles();

	const { user } = useAuthState();
	console.log(user);
	const appreciations = appreciationData.appreciation;
	return (
		<div>
			{!user ? (
				<>
					<Typography variant="h2" className="bounce-in">
						Its Graduation Day!!
					</Typography>
					<Container maxWidth="sm">
						Congratulations to all of you for successfully completing the coding bootcamp. It was a challenging journey for all of us. Sign up and join campus connect to stay in touch
						with cohort and TAs.

					</Container>
				</>
			) : (
				<>
					{' '}
					{/* <Typography>Welcome {user.data.username}</Typography>{' '} */}
					<div className={classes.root}>
						{appreciations.map(appreciation => {
							return (
								<div
									className={classes.slide}
									key={appreciation.title}>
									<Appreciate
										imageUrl={appreciation.image}
										to={appreciation.to}
										message={appreciation.message}
										from={appreciation.from}
									/>
								</div>
							)
						})}
					</div>
				</>
			)}
		</div>
	);
};

export default Home;
