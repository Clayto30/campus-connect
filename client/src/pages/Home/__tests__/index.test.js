import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Home from '../Home';

afterEach(cleanup);

describe('Home component', () => {

    it('renders', () => {
        render(<Home />);
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Home />);

        expect(asFragment()).toMatchSnapshot();
    });
})