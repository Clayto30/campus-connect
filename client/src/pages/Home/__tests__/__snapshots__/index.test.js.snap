// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    This is Home page
  </div>
</DocumentFragment>
`;
